stdlib/builtin/src/main/java/org/ballerinalang/nativeimpl/builtin/stringlib/FindAll.java

```diff
@@ -62,4 +66,19 @@ public void execute(Context context) {
             context.setReturnValues(BuiltInUtils.createStringError(context, e.getMessage()));
         }
     }
+
+    public static Object findAll(Strand strand, String value, String regex) {
+        try {
+            Pattern pattern = validatePattern(regex);
+            ArrayValue stringArray = new ArrayValue(new BArrayType(org.ballerinalang.jvm.types.BTypes.typeString));
+            Matcher matcher = pattern.matcher(value);
+            int i = 0;
+            while (matcher.find()) {
+                stringArray.add(i++, matcher.group());
+            }
+            return stringArray;
+        } catch (PatternSyntaxException e) {
+            return BallerinaErrors.createError(e.getMessage(), null);
```

Comment:
The first parameter of BallerinaErrors.createError() is the `reason`. So shall we pass builtin specific error code as the first param and e.getMessage() as the second parameter which represents the `detail`?